lib/eqaf.ml

```diff
@@ -111,3 +111,53 @@ let equal a b =
   if (al lxor ln) lor (bl lxor ln) <> 0
   then false
   else equal ~ln a b
+
+let[@inline] compare (a:int) b = a - b
+let[@inline] minus_one_or_less n = ((n land min_int) asr Sys.int_size) land 1
+let[@inline] one_if_not_zero n = (minus_one_or_less n) lor (minus_one_or_less (-n))
+
+let compare_be ~ln a b =
+  let r = ref 0 in
+  let i = ref (pred ln) in
+
+  while !i >= 0 do
+    let xa = get a !i and xb = get b !i in
+    let c = compare xa xb in
+    let n = (minus_one_or_less[@inlined]) c + (one_if_not_zero[@inlined]) c in
+    let n = n lsr (((one_if_not_zero[@inlined]) !r) lsl 1) in
+    r := n + !r ;
+    decr i ;
+  done ;
+
+  (!r land 1) - (!r lsr 1)
+
+let compare_be a b =
+  let al = String.length a in
+  let bl = String.length b in
+  let ln = min al bl in
+  if (al lxor ln) lor (bl lxor ln) <> 0
```

Comment:
If you don't want to let the programmer control `ln`, I think this will suffice, then we avoid the `min` branch?
```ocaml
if (al lxor bl) <> 0 then invalid_arg ".." else compare_be ~ln:al a b
```